packages/jitm/src/class-engine.php

```diff
@@ -0,0 +1,186 @@
+
+	/**
+	 * Returns the default rules
+	 *
+	 * @return array Default rules.
+	 */
+	public function default_rules() {
+		$rules = array_merge(
+			$this->preconnection_default_rules()
+		);
+
+		return apply_filters( 'jetpack_jitm_rules', $rules );
+	}
+
+	/**
+	 * Returns the pre-connection JITMs default rules
+	 *
+	 * @return array Pre-connection JITMs default rules.
+	 */
+	private function preconnection_default_rules() {
+
+		return array(
+			( new Message( 'jpsetup-posts', 'pre-connect' ) )
+				->user_is( 'administrator' )
+				->with_icon()
+				->message_path( '/wp:edit-post:admin_notices/' )
+				->show(
+					__( 'Do you know which of these posts gets the most traffic?', 'jetpack' ),
+					__( 'Setup Jetpack to get in-depth stats about your content and visitors.', 'jetpack' )
+				)
+				->priority( 100 )
+				->with_cta(
+					__( 'Setup Jetpack', 'jetpack' ),
+					'',
+					function() {
+						return esc_url( \Jetpack::init()->build_connect_url( true, false, 'pre-connection-jitm-posts' ) );
+					}
+				)
+				->open_cta_in_same_window()
+				->is_dismissible( true )
+				->priority( 1000 )
+				->is_hosted_with_partner( 'bluehost' ),
+			( new Message( 'jpsetup-upload', 'pre-connect' ) )
+				->user_is( 'administrator' )
+				->with_icon()
+				->message_path( '/wp:upload:admin_notices/' )
+				->show(
+					__( 'Do you want lightning-fast images?', 'jetpack' ),
+					__( 'Setup Jetpack, enable Site Accelerator, and start serving your images lightning fast, for free.', 'jetpack' )
+				)
+				->priority( 100 )
+				->with_cta(
+					__( 'Setup Jetpack', 'jetpack' ),
+					'',
+					function() {
+						return esc_url( \Jetpack::init()->build_connect_url( true, false, 'pre-connection-jitm-upload' ) );
+					}
+				)
+				->open_cta_in_same_window()
+				->is_dismissible( true )
+				->priority( 1000 )
+				->is_hosted_with_partner( 'bluehost' ),
+			( new Message( 'jpsetup-widgets', 'pre-connect' ) )
+				->user_is( 'administrator' )
+				->with_icon()
+				->message_path( '/wp:widgets:admin_notices/' )
+				->show(
+					__( 'Looking for even more widgets?', 'jetpack' ),
+					__( 'Setup Jetpack for great additional widgets like business hours and maps.', 'jetpack' )
+				)
+				->priority( 100 )
+				->with_cta(
+					__( 'Setup Jetpack', 'jetpack' ),
+					'',
+					function() {
+						return esc_url( \Jetpack::init()->build_connect_url( true, false, 'pre-connection-jitm-widgets' ) );
+					}
+				)
+				->open_cta_in_same_window()
+				->is_dismissible( true )
+				->priority( 1000 )
+				->is_hosted_with_partner( 'bluehost' ),
+		);
+	}
+
+	/**
+	 * Gets the top messages
+	 *
+	 * @param string $message_path Message path.
+	 * @param string $user_level User level.
+	 * @param string $query Message query.
+	 * @param bool   $mobile_browser Uses mobile browser.
+	 *
+	 * @return array Rendered messages.
+	 */
+	public function get_top_messages( $message_path, $user_level = '', $query = '', $mobile_browser = false ) {
```

Comment:
Since the JITMs that we're showing all have the same priority but are for different places in wp-admin, do we need to add this logic that is attempts to get the top messages?